data/src/main/java/com/example/data/implementationRepo/LastChosenCitiesRepoImpl.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.data.implementationRepo
+
+import com.example.data.db.Database
+import com.example.data.db.dao.CityDao
+import com.example.data.mappers.LastChosenCitiesEntityMapper
+import com.example.data.utils.MAX_TABLE_SIZE
+import com.example.domain.models.CityModel
+import com.example.domain.repositories.LastChosenCitiesRepo
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+
+class LastChosenCitiesRepoImpl(
+    private val cityDao: CityDao,
+    private val mapper: LastChosenCitiesEntityMapper,
+    private val database: Database
+) : LastChosenCitiesRepo {
+
+    override fun getLastChosenCities(): Single<List<CityModel>> {//This method gets list of last ten chosen cities from db and maps the list to list of CityModels
+        return cityDao.getAllLastChosenCities()
+            .subscribeOn(Schedulers.io())
```

Comment:
Let's put schedulers in order in all repos. 

Usually the repository gets schedulers in constructor. In our case all repository do operations in IO scheduler. So, this scheduler should be injected into constructors of repositories and used in subscribeOn. It is also useful for tests.
There shouldn't be subscribeOn outside of repository. UI part just uses observeOn.

To produce IO scheduler we usually use dagger annotation Named. Because there can be different Scheduler that are also produced. We need to differ them. Named is nice solution for that.

Reply:
fixed

Reply:
What about DailyWeatherRepositoryImpl?
And do a search in whole code (CTRL+SHIFT+F) to find a text subscribeOn(Schedulers.io()). There should be no one match. Everything subscribeOn() should be moved from ViewModel to Repo.